packages/providers/src/core/PoapCollectionsApi/PoapCollectionsApi.ts

```diff
@@ -0,0 +1,153 @@
+import { CollectionsUnauthorizedError } from './../../ports/CollectionsApiProvider/errors/CollectionsUnauthorizedError';
+import { CollectionsBadRequestError } from '../../ports/CollectionsApiProvider/errors/CollectionsBadRequestError';
+import { PostCollectionsInput } from '../../ports/CollectionsApiProvider/types/PostCollectionsInput';
+import { CollectionResponse } from '../../ports/CollectionsApiProvider/types/CollectionResponse';
+import { PutCollectionsInput } from '../../ports/CollectionsApiProvider/types/PutCollectionsInput';
+import { CollectionsApiProvider } from '../../ports/CollectionsApiProvider/CollectionsApiProvider';
+
+const DEFAULT_COLLECTIONS_BASE_URL = 'https://collections.poap.tech';
+/**
+ * The `PoapCollectionsApi` class provides methods to interact with the POAP Collections API,
+ * allowing for the creation and modification of POAP collections.
+ *
+ * It requires an authToken for authorization and uses a provided or default base URL for API requests.
+ *
+ * @class PoapCollectionsApi
+ * @implements {CollectionsApiProvider}
+ */
+export class PoapCollectionsApi implements CollectionsApiProvider {
```

Comment:
I think a better name `RestApiCollectionsApi`

Reply:
Why is better to put the word REST there? None of the other providers have it. I already dislike a little that is done in REST speak instead of business speak. But hey, do it and try to fit the REST naming as far as you can instead of coming with a ubiquitous language for your business implementations.